frontend/src/pages/question/QuestionNew.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useContext } from 'react';
+import {
+  Grid, Row,
+} from 'react-bootstrap';
+import { useHistory } from 'react-router-dom';
+
+import UserContext from '@/context/user';
+import AlertContext from '@/context/alert';
+
+import API from '@/API';
+
+import Button from '@material-ui/core/Button';
+
+import './newQuestion.css';
+
+import queryGraphUtils from '@/utils/queryGraph';
+import { useVisibility } from '@/utils/cache';
+
+import QuestionBuilder from './questionBuilder/QuestionBuilder';
+
+import useQuestionStore from './useQuestionStore';
+
+export default function QuestionNew() {
+  const history = useHistory();
+  const visibility = useVisibility();
+  const user = useContext(UserContext);
+  const questionStore = useQuestionStore();
+  const displayAlert = useContext(AlertContext);
+
+  function onDownloadQuestion() {
+    const query_graph = queryGraphUtils.convert.internalToReasoner(
+      questionStore.query_graph,
+    );
+    const data = { query_graph };
+
+    // Transform the data into a json blob and give it a url
+    const json = JSON.stringify(data, null, 2);
+    const blob = new Blob([json], { type: 'application/json' });
+    const url = URL.createObjectURL(blob);
+
+    // Create a link with that URL and click it.
+    const a = document.createElement('a');
+    a.download = 'robokopMachineQuestion.json';
+    a.href = url;
+    a.click();
+    a.remove();
+  }
+
+  function onResetQuestion() {
+    questionStore.resetQuestion();
+  }
+
+  async function getAnswerAndDisplayAlert(questionId) {
+    let response = await API.queryDispatcher.getAnswer(questionId, user.id_token);
+    if (response.status === 'error') {
+      displayAlert('error', response.message);
+      return;
+    }
+    const newAnswerId = response.id;
+
+    response = await API.cache.getQuestion(questionId, user.id_token);
+    if (response.status === 'error') {
+      displayAlert('error', response.message);
+      return;
+    }
+    const questionMeta = response;
+    questionMeta.metadata.hasAnswers = true;
+    response = await API.cache.updateQuestion(questionMeta, user.id_token);
+    if (response.status === 'error') {
+      displayAlert('error', response.message);
+      return;
+    }
+
+    displayAlert(
```

Comment:
This seems like bad UX. We shouldn't make the user click this alert when the answer comes back. And we shouldn't show errors on the questionAnswerViewer page while we wait for an answer. I think when a user submits a question, we can go to the questionAnswerViewer page and show the alert saying we're getting an answer and then just show a loading icon. Once we get the answer, we should automatically show it.